src/com/google/enterprise/adaptor/filenet/DocumentTraverser.java

```diff
@@ -322,69 +327,56 @@ private void processInheritedPermissions(DocId docId,
       ActiveMarkingList activeMarkings, Permissions.Acl permissions,
       Response response) {
     // Send add request for adding ACLs inherited from parent folders.
```

Comment:
Er, move this comment down? Add a comment here that active marking sets must be ANDed with the other permissions, and the GSA only supports AND_BOTH_PERMIT (at least, with the correct semantics) at the root of the inheritance chain.